okio/src/nativeTest/kotlin/okio/DeflaterTest.kt

```diff
@@ -23,177 +23,237 @@ import kotlin.test.assertTrue
-    assertFalse(deflater.process())
-    assertEquals(deflater.targetLimit, deflater.targetPos)
-    targetBuffer.write(deflater.target)
-
-    deflater.target = ByteArray(256)
-    deflater.targetPos = 0
-    deflater.targetLimit = deflater.target.size
-    assertTrue(deflater.process())
-    assertEquals(deflater.sourcePos, deflater.sourceLimit)
-    targetBuffer.write(deflater.target, 0, deflater.targetPos)
-
-    // Golden compressed output.
-    assertEquals(
-      "c89PUchIzSlQKC3WUShPVS9KVcjMUyjJSFXISMxLKVbIT1NIzUvPzEtNLSrWAwA=".decodeBase64(),
-      targetBuffer.readByteString(),
-    )
-
-    deflater.close()
+      target = ByteArray(10)
+      targetPos = 0
+      targetLimit = target.size
+      assertFalse(process())
+      assertEquals(targetLimit, targetPos)
+      assertEquals(content.length.toLong(), deflater.getBytesRead())
+      targetBuffer.write(target)
+
+      target = ByteArray(256)
+      targetPos = 0
+      targetLimit = target.size
+      assertTrue(process())
+      assertEquals(sourcePos, sourceLimit)
+      assertEquals(content.length.toLong(), deflater.getBytesRead())
+      targetBuffer.write(target, 0, targetPos)
+
+      // Golden compressed output.
+      assertEquals(
+        "eJxzz09RyEjNKVAoLdZRKE9VL0pVyMxTKMlIVchIzEspVshPU0jNS8/MS00tKtYDAF6CD5s=".decodeBase64(),
+        targetBuffer.readByteString(),
+      )
+
+      deflater.end()
+    }
   }
 
   @Test
   fun deflateEmptySource() {
-    val deflater = Deflater().apply {
-      flush = Z_FINISH
+    val deflater = Deflater()
+    deflater.dataProcessor.apply {
+      deflater.flush = Z_FINISH
 
       target = ByteArray(256)
       targetPos = 0
       targetLimit = target.size
-    }
 
-    assertTrue(deflater.process())
-    val deflated = deflater.target.toByteString(0, deflater.targetPos)
+      assertTrue(process())
+      assertEquals(0L, deflater.getBytesRead())
+      val deflated = target.toByteString(0, targetPos)
+
+      // Golden compressed output.
+      assertEquals(
+        "eJwDAAAAAAE=".decodeBase64(),
+        deflated,
+      )
+
+      deflater.end()
+    }
+  }
 
-    // Golden compressed output.
-    assertEquals(
-      "AwA=".decodeBase64(),
-      deflated,
-    )
+  @Test
+  fun cannotDeflateAfterEnd() {
+    val deflater = Deflater()
+    deflater.end()
 
-    deflater.close()
+    assertFailsWith<IllegalStateException> {
+      deflater.dataProcessor.process()
+    }
   }
 
   @Test
-  fun cannotDeflateAfterClose() {
+  fun cannotGetBytesReadAfterEnd() {
     val deflater = Deflater()
-    deflater.close()
+    deflater.end()
 
     assertFailsWith<IllegalStateException> {
-      deflater.process()
+      deflater.getBytesRead()
     }
   }
 
   @Test
-  fun closeIsIdemptent() {
+  fun endIsIdemptent() {
     val deflater = Deflater()
-    deflater.close()
-    deflater.close()
+    deflater.end()
+    deflater.end()
   }
+
+  private fun Deflater(nowrap: Boolean) = Deflater(Z_BEST_COMPRESSION, nowrap)
```

Comment:
No way to put this default value onto the expect so you don't need to define this function?

Reply:
Not as far as I know.

```
> Task :okio:compileKotlinJvm FAILED
e: file:///Volumes/Development/okio/okio/src/jvmMain/kotlin/okio/-JvmPlatform.kt:45:1 Default argument values inside expect declaration 'Deflater' are not allowed if it is actualized via typealias. Possible fix is to remove default argument values in members:
    public constructor Deflater(level: Int = ..., nowrap: Boolean = ...)
```